deploy.ps1

```diff
@@ -0,0 +1,165 @@
+        Name     = "Keyvault Secrets"
+        Describe = "Inject Secrets into Session"
+        Set      = {
+            $KEYVAULTNAME = $kv_name
+            $SECRETS = ( $(az keyvault secret list --vault-name $KEYVAULTNAME | jq '.[].id' -r | sed 's/.*\/\([^/]\+\)$/\1/') )
+            $SECRETS | % {
+                $SECRET = $(az keyvault secret show --name $_ --vault-name $KEYVAULTNAME | jq '.value' -r)
+                $NAME = $_.Replace("-", "_")
+                [Environment]::SetEnvironmentVariable($NAME, $SECRET)
+            }
+        }
+    }
+)
+
+# Provision Infra
+$tfReqs = @(
+    @{
+        Name     = "Terraform init"
+        Describe = "Initialize terraform environment"
+        Test     = { Test-Path "$PSScriptRoot/tf/.terraform" }
+        Set      = {
+            Set-Location -Path "tf"
+            terraform init -backend-config="storage_account_name=$($tf_share)" `
+                -backend-config="container_name=tfstate" `
+                -backend-config="access_key=$($env:terraform_storage_key)" `
+                -backend-config="key=mics.tfstate"
+        }
+    },
+    @{
+        Name     = "Terraform plan"
+        Describe = "Plan terraform environment"
+        Test     = { Test-Path "$OutputDir/out.plan" }
+        Set      = {
+            New-Item -Path "$OutputDir" -ItemType Directory -Force
+            terraform plan -out "$OutputDir/out.plan"
+        }
+    },
+    @{
+        Name     = "Terraform Apply"
+        Describe = "Apply Terraform plan"
+        Test     = { Test-Path "$OutputDir/azurek8s" }
+        Set      = {
+            terraform apply "$OutputDir/out.plan" | Write-Information
+            terraform output kube_config | Out-File "$OutputDir/azurek8s"
+            Set-Location $RepoRoot
+        }
+    }
+)
+
+# Docker cooking
+$dockerReqs = @(
+    @{
+        Name     = "Find Docker Services"
+        Describe = "Enumerate Containers"
+        Set      = {
+            Set-k8sConfig -AppPath "./app" -OutPath "./out"
+        }
+    },
+    @{
+        Name     = "Build Docker Containers"
+        Describe = "Build all containers"
+        Set      = {
+            $list = Get-Content ./out/k8s.json | ConvertFrom-Json
+            $list | % { docker build -t "$acr_name/$($_.ImageName)" -f $_.Name $_.Path }
+        }
+    },
+    @{
+        Name     = "Push Containers"
+        Describe = "Push all containers"
+        Set      = {
+            docker login $acr_name -u mics233 -p $env:acrpassword
+
+            $list = Get-Content ./out/k8s.json | ConvertFrom-Json
+            $list | % { docker push "$acr_name/$($_.ImageName)" }
+        }
+    }
+)
+
+# Kubernetes Deployment
+$k8sReqs = @(
+    @{
+        Name     = "Load Config"
+        Describe = "Load k8s config"
+        Set      = {
+            $env:KUBECONFIG = "./out/azurek8s"
+        }
+    },
+    @{
+        Name     = "Generate pod.yml"
+        Describe = "Generate pod.yml"
```

Comment:
Pod.yml defines what runs in kubernetes.  We're generating one from templates and making it reference the container we build and pushed up to our container registry.